addons/netfox.extras/state-machine/network-state-machine.gd

```diff
@@ -0,0 +1,51 @@
+@icon("res://addons/netfox/icons/state-synchronizer.svg")
+extends Node
+class_name NetworkedStateMachine
+
+## A networked state machine that can be used to synchronize state between peers.
+
+@export var state: StringName = "": set = set_state
+
+signal state_changed(old_state: NetworkedState, new_state: NetworkedState)
+
+# Semi-private state property
+var _state_object: NetworkedState = null
+var _available_states: Dictionary = {}
+
+# Use the ready function to gather all available states
+func _ready():
+	for child in get_children():
+		if child is NetworkedState:
+			_available_states[child.name] = child
+
+# Method to set the state
+func set_state(new_state: StringName) -> void:
+	if !new_state:
+		return
+	
+	if state == new_state:
+		return
+	
+	if !_available_states.has(new_state):
+		return
+		
+	var new_state_class: NetworkedState = _available_states[new_state]
+	if !new_state_class:
+		return
+	
+	if _state_object:
+		
+		if !new_state_class.can_enter(_state_object):
+			return
```

Comment:
What do you think about a `push_warning` here? I assume people don't intentionally code invalid transitions, so users could use a warning about it.

Reply:
In the example, `can_enter` checks if the player is on the floor before transitioning to the jump state. In this case, I think a warning wouldn't be useful.

Reply:
True, fair enough!